plugin/validator_test.go

```diff
@@ -956,6 +956,889 @@ func TestValidator_SchemaAsJSON(t *testing.T) {
+				"host_with_optional_port": "192.0.2.1.1:80",
+				"tag": "test"
+			}`,
+			wantErr:     true,
+			expectedErr: `{"config":{"host_with_optional_port":"invalid ipv4 address: 192.0.2.1.1:80"}}`,
+		},
+		{
+			name: "valid wildcard_host",
+			config: `{
+				"wildcard_host": "*test.com",
+				"tag": "test"
+			}`,
+		},
+		{
+			name: "valid wildcard_host",
+			config: `{
+				"wildcard_host": "test.*",
+				"tag": "test"
+			}`,
+		},
+		{
+			name: "invalid wildcard_host with multiple *",
+			config: `{
+				"wildcard_host": "*test.*",
+				"tag": "test"
+			}`,
+			wantErr:     true,
+			expectedErr: `{"config":{"wildcard_host":"only one wildcard must be specified"}}`,
+		},
+		{
+			name: "invalid wildcard_host with * in the middle",
+			config: `{
+				"wildcard_host": "new*test",
+				"tag": "test"
+			}`,
+			wantErr:     true,
+			expectedErr: `{"config":{"wildcard_host":"wildcard must be leftmost or rightmost character"}}`,
+		},
+		{
+			name: "valid ipv4",
+			config: `{
+				"ip": "192.0.2.1",
+				"tag": "test"
+			}`,
+		},
+		{
+			name: "invalid ipv4",
+			config: `{
+				"ip": "192.0.2.1.1",
+				"tag": "test"
+			}`,
+			wantErr:     true,
+			expectedErr: `{"config":{"ip":"not an ip address: 192.0.2.1.1"}}`,
+		},
+		{
+			name: "valid ipv6",
+			config: `{
+				"ip": "2001:db8::1",
+				"tag": "test"
+			}`,
+		},
+		{
+			name: "invalid ipv6",
+			config: `{
+				"ip": "2001:db8::1::1",
+				"tag": "test"
+			}`,
+			wantErr:     true,
+			expectedErr: `{"config":{"ip":"not an ip address: 2001:db8::1::1"}}`,
+		},
+		{
+			name: "valid ip_or_cidr",
+			config: `{
+				"ip_or_cidr": "192.0.2.0/24",
+				"tag": "test"
+			}`,
+		},
+		{
+			name: "invalid ip_or_cidr with too big mask",
+			config: `{
+				"ip_or_cidr": "192.0.2.0/244",
+				"tag": "test"
+			}`,
+			wantErr:     true,
+			expectedErr: `{"config":{"ip_or_cidr":"invalid ip or cidr range: '192.0.2.0/244'"}}`,
+		},
+		{
+			name: "valid cidr_v4",
+			config: `{
+				"cidr_v4": "192.0.2.0/24",
+				"tag": "test"
+			}`,
+		},
+		{
+			name: "invalid cidr_v4 with too big mask",
+			config: `{
+				"cidr_v4": "192.0.2.0/244",
+				"tag": "test"
+			}`,
+			wantErr:     true,
+			expectedErr: `{"config":{"cidr_v4":"invalid ipv4 cidr range: '192.0.2.0/244'"}}`,
+		},
+		{
+			name: "valid port",
+			config: `{
+				"port": 80,
+				"tag": "test"
+			}`,
+		},
+		{
+			name: "invalid port as string",
+			config: `{
+				"port": "80",
+				"tag": "test"
+			}`,
+			wantErr:     true,
+			expectedErr: `{"config":{"port":"expected an integer"}}`,
+		},
+		{
+			name: "invalid too big port",
+			config: `{
+				"port": 9999999999,
+				"tag": "test"
+			}`,
+			wantErr:     true,
+			expectedErr: `{"config":{"port":"value should be between 0 and 65535"}}`,
+		},
+		{
+			name: "invalid negative port",
+			config: `{
+				"port": -1,
+				"tag": "test"
+			}`,
+			wantErr:     true,
+			expectedErr: `{"config":{"port":"value should be between 0 and 65535"}}`,
+		},
```

Comment:
Let's try the hard edge of 0 and 65535 as well; WDYT?

Reply:
https://github.com/Kong/goks/pull/14/commits/035eab33cac54f174d8195c8f8689d79327600d1